### PR TITLE
limit verbosity to supported range

### LINKED ISSUE #1283: Unhandled exception with multiple `--quiet` flags
**Describe the bug**
Giving `-q` more than twice to commands like `list` gives unhandled exception:

```
$ pipx list -qqq
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/logging/config.py", line 564, in configure
    handler = self.configure_handler(handlers[name])
  File "/usr/local/lib/python3.9/logging/config.py", line 758, in configure_handler
    result.setLevel(logging._checkLevel(level))
  File "/usr/local/lib/python3.9/logging/__init__.py", line 198, in _checkLevel
    raise ValueError("Unknown level: %r" % level)
ValueError: Unknown level: 'Level 60'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/pipx", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.9/site-packages/pipx/main.py", line 916, in cli
    setup(parsed_pipx_args)
  File "/usr/local/lib/python3.9/site-packages/pipx/main.py", line 856, in setup
    setup_logging(verbose)
  File "/usr/local/lib/python3.9/site-packages/pipx/main.py", line 846, in setup_logging
    logging.config.dictConfig(logging_config)
  File "/usr/local/lib/python3.9/logging/config.py", line 809, in dictConfig
    dictConfigClass(config).configure()
  File "/usr/local/lib/python3.9/logging/config.py", line 571, in configure
    raise ValueError('Unable to configure handler '
ValueError: Unable to configure handler 'stream'
```

**How to reproduce**

`pipx list -qq` works fine, `pipx list -qqq` does not.

**Expected behavior**

Either ignore the superfluous `q`-flags, or print a notice about the number of times this subcommand accepts the flag.

### PR DESCRIPTION
**Pull request by Gitznik**
- [x] I have added a news fragment under `changelog.d/` (if the patch affects the end users)

## Summary of changes
Fixes #1283 by clamping the possible verbosity value.

## Test plan

Tested by running

```
pipx list -qqq
pipx list -vvv
```

### CODE REVIEW

File: src/pipx/main.py [hunk 1]
```diff
@@ -758,7 +758,7 @@ def get_command_parser() -> Tuple[argparse.ArgumentParser, Dict[str, argparse.Ar
         default=0,
         help=(
             "Give less output. May be used multiple times corresponding to the"
-            " WARNING, ERROR, and CRITICAL logging levels."
+            " ERROR, and CRITICAL logging levels. Any count > 2 is ignored."
```
Comment (chrysle) [inline src/pipx/main.py:761]:
```suggestion
            " ERROR and CRITICAL logging levels. Any count > 2 is ignored."
```

**Review by chrysle — DISMISSED** [review]
Thank you!

**Comment by chrysle** [comment]
Can you amend the help for the `--verbose` flag, too?

File: src/pipx/main.py (hunk 1 above)
Comment (Gitznik) [reply src/pipx/main.py]: Done :+1:

File: src/pipx/main.py [hunk 2]
```diff
@@ -758,11 +758,20 @@ def get_command_parser() -> Tuple[argparse.ArgumentParser, Dict[str, argparse.Ar
...
 
-    shared_parser.add_argument("--verbose", "-v", action="count", default=0, help=("Give more output."))
+    shared_parser.add_argument(
+        "--verbose",
+        "-v",
+        action="count",
+        default=0,
+        help=(
+            "Give more output. May be used multiple times corresponding to the"
+            " INFO, DEBUG and NOTSET logging levels. Any count > 3 is ignored."
```
Comment (chrysle) [inline src/pipx/main.py:772]: According to my tests, exceeding the count will also result in the `NOTSET` logging level.

```suggestion
            "Give more output. May be used multiple times corresponding to the"
            " INFO, DEBUG and NOTSET logging levels."
```

Comment (Gitznik) [reply src/pipx/main.py:772]: Ah I see how the wording is misleading. Maybe like this:

Comment (chrysle) [reply src/pipx/main.py:772]: Sounds good!

Comment (Gitznik) [reply src/pipx/main.py]: done

File: changelog.d/1283.bugfix.md [hunk 3]
```diff
@@ -0,0 +1 @@
+Limit the number of possible -q and -v by ignoring exceeding values
```
Comment (chrysle) [inline changelog.d/1283.bugfix.md:1]:
```suggestion
Fix crashes due to superfluous `-q ` and `-v` flags by discarding exceeding values
```

Comment (Gitznik) [reply changelog.d/1283.bugfix.md:1]: superfluous `-v` flags were already discarded. Will rewrite it for `-q`

**Review by chrysle — APPROVED** [review]

**Pull request merged**